pkg/thanos/statefulset.go

```diff
@@ -303,13 +327,37 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 			watchedDirectories = append(watchedDirectories, mountPath)
 		}
 
+		var fields monitoringv1.WebConfigFileFields
+		if tr.Spec.Web != nil {
+			fields = tr.Spec.Web.WebConfigFileFields
+		}
+
+		webConfig, err := webconfig.New(webConfigDir, webConfigSecretName(tr.Name), fields)
+		if err != nil {
+			return nil, err
+		}
+
+		confArg, configVol, configMount, err := webConfig.GetMountParameters()
+		if err != nil {
+			return nil, err
+		}
+		containerArgs = append(containerArgs, fmt.Sprintf("--%s=%s", "http.config", confArg.Value))
```

Comment:
we need a version check similar to https://github.com/prometheus-operator/prometheus-operator/blob/984b6e9697977fb494b006de3f20da4abc1d7a24/pkg/prometheus/server/statefulset.go#L277-L309